dev/doc/changes.md

```diff
@@ -1,5 +1,17 @@
 ## Changes between Coq 8.14 and Coq 8.15
 
+### Ltac Visual Debugger in CoqIDE
+
+- CoqIDE now supports setting breakpoints visually in Ltac tactics.  When
+  the debugger stops (e.g. at a breakpoint), CoqIDE highlights the next
+  tactic to execute in the script buffer.  Execution can be resumed with
+  continue, step over, step in and step out operations.  Displays the call
+  stack and variable values for each stack frame in a new panel.
+- Updated the XML protocol (4 new "db_*" messages and added location info
+  to the "add" message) and added support for these primarily in tactic_debug.ml
+- Currently only for Ltac in CoqIDE, but readily extendable to other
+  scripting languages and IDEs.
+
```

Comment:
I think items 1 and 3 may not be relevant in this file, as they are more like user facing changes? [And already in the main changelog I guess]

I would maybe remove them, and update item 2 to specify the relevant change to add in more detail. Something such as: "before this change, add returned `<xml> .... </xml>` , now, new attributes `<foo> ...` and `<bar>` have been added.

Reply:
Or provide a pointer to the document where the new return format is spec'ed.

Reply:
Hard to know what should and shouldn't appear in this file.  I removed items 1 and 3.  I added a pointer to `xmlprotocol.md`, which has detailed descriptions of all the messages and responses.  Note that "add" got extra parameters rather than extra return values.

Reply:
Note that this PR includes an overlay for coqtail to address this very point.  Aside from the PG prototype, are you aware of any other packages that use the xml protocol?

Reply:
There are some other users indeed, not very actively maintained, but given what we wrote on the protocol file we need to signal the protocol changed [or introduce backwards compatible changes]

Personally I'd keep the changelog for the protocol in the .mli file, in a format `-date-protocol: changes`

Reply:
Another good practice for this kind of changelogs is to signal explicitly that the change is backwards incompatible, thus you can write:
```
# XML protocol (IMHO better under its own header)
- The add message will now return extra fields for location of the added sentence (warning, backwards-incompatible change, see protocol.md file for more details)
- 4 new "db_*" messages have been added to support the ltac_debugger
```